lib/SILGen/SILGenType.cpp

```diff
@@ -59,8 +59,13 @@ SILGenModule::emitVTableMethod(ClassDecl *theClass,
   // case we won't emit it if building a resilient module.
   SILVTable::Entry::Kind implKind;
   if (baseClass == theClass) {
-    // This is a vtable entry for a method of the immediate class.
-    implKind = SILVTable::Entry::Kind::Normal;
+    if (derivedDecl->isResilientFinal())
```

Comment:
Can you reuse `isNonOverridden` here? In the full generality I ended up needing that to be orthogonal to the existing set of kinds.

Reply:
Can you remember why?  The modeling seems *much* cleaner this way.

Reply:
I don't completely remember the specifics, but the distinction between `Normal` and `Inherited` is significant, and when a method has no overrides but the class does have other subclasses, then it is perfectly valid to have an entry that is both inherited and non-overridden in the subclass.